setup.cfg

```diff
@@ -46,7 +46,7 @@ install_requires =
     nbformat
     packaging
     psutil
-    pulp >=2.3.1,<2.9 # pulp introduced a breaking change from 2.7 to 2.8. Hence we should always pin to the minor version.
+    pulp >=2.0
```

Comment:
Mhm, I get the idea, but who knows what pulp will break next on a minor release. Check poetry, it is nowadays default behavior to be conservative with dependency pinning. I think we should gradually do that for all dependencies and rather rely on dependabot for updating them.